src/shared/tpm2-util.c

```diff
@@ -596,8 +598,65 @@ static int tpm2_get_best_pcr_bank(
         return 0;
 }
 
+static int tpm2_make_hmac_session(
```

Comment:
i wonder if this should be called tpm2_setup_parameter_encryption() or so? i.e. focus on the purpose, not on the implementation detail?